### PR TITLE
[wip] event logging

### PR DESCRIPTION
**Pull request by kazuho**
* [x] in default_event_log, check for buffer overrun
* [x] log current time
* [x] log all CC events
* [x] log CID
* [ ] log remote address

### CODE REVIEW

**Comment by kazuho** [comment]
@janaiyengar I think we might need to discuss two issues.

* If we have enough information in the logs.
* If we want to combine major events and minor events into one object. We currently log major events (i.e., connect, accept, send, receive) that has timestamp, CID as attributes, then log other (i.e. "minor") events that happen as part of the major event. At the moment, each event is logged as distinct JSON objects. However, we can change the design so that a major event object will contain a list of minor events.

**Comment by janaiyengar** [comment]
Thank you for working on this Kazuho! I really like the high-level design, and the use of a mask to filter specific logs. I've gone over it once, and just a couple of clarification questions. But I want to go over the log format itself in more detail, which I'll do tomorrow, and we can talk about it in our meeting tomorrow. High-order point: Can you move your review comments to the code itself as you note in your comment above?

File: lib/quicly.c [hunk 1]
```diff
@@ -2760,18 +2760,17 @@ static int handle_ack_frame(quicly_conn_t *conn, size_t epoch, quicly_ack_frame_
     }
     if (cc_type != 0)
         cc_cong_signal(&conn->egress.cc.ccv, cc_type, bytes_in_flight);
-    int end_of_recovery = frame->largest_acknowledged >= conn->egress.cc.end_of_recovery;
+    int exit_recovery = frame->largest_acknowledged >= conn->egress.cc.end_of_recovery;
```
Comment (janaiyengar) [inline lib/quicly.c]: Is end-of-recovery also being populated somewhere?

File: include/quicly.h [hunk 2]
```diff
@@ -100,6 +164,10 @@ struct st_quicly_context_t {
      * tls context to use
      */
     ptls_context_t tls;
+    /**
+     *
+     */
+    uint64_t next_master_id;
```
[the remark this answers is not in the document]
Comment (kazuho) [reply include/quicly.h]: master_id designates a QUIC connection (note: QUIC CIDs designate a flow of a QUIC connection).
`next_master_id` is a field used for assigning master_id to each of the connection. Value of `next_master_id` is incremented every once the value is assigned to a new connection.

File: lib/quicly.c (hunk 1 above)
Comment (kazuho) [reply lib/quicly.c]: `end_of_recovery` is the PN that marks the end of recovery. `exit_recovery` is a boolean indicating if the congestion controller is exiting recovery.

The two values are logged under the same names. `end-of-recovery` is logged when congestion is detected. `exit-recovery` is logged when a new ACK is received.

**Comment by kazuho** [comment]
The CLI command logs events to file descriptor 5 (if it is open). As an example, you can do `cli ... 5> events.log` to log them.

**Pull request merged**